src/clj/rems/db/user_mappings.clj

```diff
@@ -14,23 +14,53 @@
 
 (defn- format-user-mapping [mapping]
   {:userid (:userid mapping)
-   :ext-id-attribute (:extidattribute mapping)
+   :ext-id-attribute (keyword (:extidattribute mapping))
    :ext-id-value (:extidvalue mapping)})
 
-(defn get-user-mappings
-  [params]
-  (->> (db/get-user-mappings (map-vals name params))
+(defn- load-user-mappings []
+  (->> (db/get-user-mappings {})
        (mapv format-user-mapping)
        (mapv validate-user-mapping)
+       (group-by :ext-id-value)))
+
+;; NB: user mappings are always cached
+;; XXX: consider if this should rather be mount state eventually?
+(def ^:private user-mappings-by-value (atom nil))
+
+(defn- ensure-cached! []
+  (when (nil? @user-mappings-by-value)
+    (reset! user-mappings-by-value (load-user-mappings))))
```

Comment:
now that we are caching user mappings in-memory, should we remove the index on the user mappings db table? it is probably unnecessary use of db resources https://github.com/CSCfi/rems/blob/master/resources/migrations/20220120121117-user-mappings.up.sql

Reply:
We could do that now and then we fix only the migration here and not introduce a new one.